### PR TITLE
refactor: migrate to oclif/core

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?
Migrates config commands to oclif/core

### What issues does this PR fix or reference?
@W-9641000@

### CODE REVIEW

File: src/commands/config/set.ts [hunk 1]
```diff
@@ -42,25 +42,59 @@ export class Set extends ConfigCommand {
...
+        throw messages.createError('InvalidArgumentFormat', [arg]);
+      }
+
+      const [name, value] = split;
+
+      if (configs[name]) {
+        throw messages.createError('DuplicateArgument', [name]);
+      }
+
+      configs[name] = value || undefined;
```
Comment (peternhale) [inline src/commands/config/set.ts:89]: @mdonnalley, is this to support backward compatibility to a time where command `config:unset` was not available?

File: src/commands/config/set.ts [hunk 2]
```diff
@@ -42,25 +42,59 @@ export class Set extends ConfigCommand {
...
+  protected async resolveArguments(): Promise<string[]> {
+    const { args, argv } = await this.parse(Set);
+
+    const argVals = Object.values(args);
+    return argv.filter((val) => !argVals.includes(val));
+  }
+
+  protected async parseConfigKeysAndValues(): Promise<{ [index: string]: string }> {
+    const configs: { [index: string]: string } = {};
+    const args = await this.resolveArguments();
```
Comment (peternhale) [inline src/commands/config/set.ts:64]: Maybe use [for await...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of)

**Comment by RodEsp** [comment]
Should we close https://github.com/salesforcecli/plugin-config/pull/162in favor of this PR?

File: src/config.ts [hunk 3]
```diff
@@ -5,84 +5,84 @@
...
+  }
+
+  protected pushFailure(name: string, err: string | Error, value?: string): void {
+    const error = SfdxError.wrap(err);
+    this.responses.push({
+      name,
+      success: false,
+      value,
+      error,
+      message: error.message.endsWith('..') ? error.message.replace('..', '.') : error.message,
```
Comment (peternhale) [inline src/config.ts:45]: @mdonnalley what about a message that has `..some text here..`? The replace results will be `.some text here..`

Comment (peternhale) [reply src/config.ts:45]: Since the string has to be checked every time, why not use
``` suggest
message: error.message.replace(/\.\.$/, '.'),
```

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/commands/config/set.ts (hunk 1 above)
Comment (mdonnalley) [reply src/commands/config/set.ts:89]: That's correct

**Review by peternhale — APPROVED** [review]

**Pull request merged**